factorGraphEstimator:
  debug: true
  startTime: 0.0
  isamParameters:
    relinearizeThreshold: 0.01
    relinearizeSkip: 1
    enablePartialRelinearizationCheck: false
    chacheLinearedFactors: false
    enableDetailedResults: true
    findUnusedFactorSlots: false
    gaussianWildfireThreshold: 0.001

  imuFactorParams:
    # general
    useImuFactor: true
    imuBiasIncr: 5
    invertX: false
    invertY: false
    invertZ: false

    # factor specific
    accelNoiseSigma: 0.1
    gyroNoiseSigma: 0.01
    accelBiasRwSigma: 0.01
    gyroBiasRwSigma: 0.01
    integrationErrorCov: 1e-4
    biasAccOmegaInt: 0.1

    # bias noise
    biasNoise: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1]

  poseFactorParams:
    usePoseFactor: true
    # rad, rad, rad, m, m, m
    poseNoise: [1.0, 1.0, 1.0, 5.0, 5.0, 5.0]
    poseVelNoise: [5.0, 5.0, 5.0]

  cameraFactorParams:
    useCameraFactor: true
    defaultPixelNoise: 10.0

    cameraNames: [left_left, left_right, right_left, right_right]

    # list of known objects to track, no - allowed
    objects: [AIRRLogo, GateCorner]

    AIRRLogo_noise: 5
    GateCorner_noise: 10

    left_left:
      # fx, fy, s, u0, v0
      K: [548.4088134765625, 548.4088134765625, 0.0, 512.0, 384.0]

      # transform x, y, z, qw, qx, qy, qz
      transform: [0.0, 0.0, 0.0, -0.5, 0.5, -0.5, 0.5]

    left_right:
      # fx, fy, s, u0, v0
      K: [548.4088134765625, 548.4088134765625, 0.0, 512.0, 384.0]

      # transform x, y, z, qw, qx, qy, qz
      transform: [0.0, 0.0, 0.0, -0.5, 0.5, -0.5, 0.5]

    right_left:
      # fx, fy, s, u0, v0
      K: [548.4088134765625, 548.4088134765625, 0.0, 512.0, 384.0]

      # transform x, y, z, qw, qx, qy, qz
      transform: [0.0, 0.0, 0.0, -0.5, 0.5, -0.5, 0.5]

    right_right:
      # fx, fy, s, u0, v0
      K: [548.4088134765625, 548.4088134765625, 0.0, 512.0, 384.0]

      # transform x, y, z, qw, qx, qy, qz
      transform: [0.0, 0.0, 0.0, -0.5, 0.5, -0.5, 0.5]


  priorConfig:
    # x, y, z, qw, qx, qy, qz, x_dor, y_dot, z_dot
    initial_state: [0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    initial_vel_noise: 0.1
    # rad, rad, rad, m, m, m
    initial_pose_noise: [0.05, 0.05, 0.05, 0.25, 0.25, 0.25]
    initial_bias_noise: 0.1

